src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -33,22 +33,25 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
-  public boolean fire(int numberOfTorpedos) throws NoSuchAlgorithmException{
+  public boolean fire(int numberOfTorpedos) {
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
       throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
-
-
     // simulate random overheating of the launcher bay which prevents firing
    
-    Random rand = SecureRandom.getInstanceStrong();
+    Random rand = new SecureRandom();
+    try {
+        rand = SecureRandom.getInstanceStrong();
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
```

Comment:
seems like this way of showing there is an error has some security problems. Here is a solution that you might use instead.
<img width="903" alt="Screenshot 2021-04-28 at 18 07 24" src="https://user-images.githubusercontent.com/27647952/116436465-b9049c00-a84c-11eb-9422-3c678b191869.png">